libs/langchain/langchain/memory/chat_message_histories/in_memory.py

```diff
@@ -19,5 +19,8 @@ def add_message(self, message: BaseMessage) -> None:
         """Add a self-created message to the store"""
         self.messages.append(message)
 
-    def clear(self) -> None:
-        self.messages = []
+    def clear(self,k=0) -> None:
+        if k:
+            del(self.messages[:-(k*2)])
```

Comment:
- yes, in paperqa, first we append the last q+a, then call clear(), which deletes everything but last k q+a
- even if the list has less than k items, del() would maintain this. It only deletes if list becomes > k
- k*2 because the list is storing both q and a as separate elements, so if k=2 we delete 4 elements from list(2q + 2 a)

Reply:
In clear(), it might be better to check that list > k elements before deleting the items.